refarch-cli/index.ts

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env node
+import { cpSync, renameSync, rmSync } from "fs";
+
+import { confirm, input, select, Separator } from "@inquirer/prompts";
+import { replaceInFileSync } from "replace-in-file";
+
+enum applications {
+  FRONTEND = "frontend",
+  BACKEND = "backend",
+  EAI = "eai",
+}
+const EXIT = "exit";
+let keepStack = true;
+let keepDocs = true;
+let hasJavaApplicationBeenGenerated = false;
+
+/**
+ * Starting-point for the Projekt configuration.
+ * Depending on the value of the select, an interactive CLI will be shown
+ */
+async function projectConfiguration() {
+  await select({
+    message: "Select Project/s you want to generate with space",
+    choices: [
+      { name: applications.FRONTEND, value: applications.FRONTEND },
+      { name: applications.BACKEND, value: applications.BACKEND },
+      { name: applications.EAI, value: applications.EAI },
+      { name: EXIT, value: EXIT },
+      new Separator(),
+    ],
+  }).then(async (result: string) => {
+    switch (result) {
+      case applications.EAI:
+        await generateJavaInteractiveCli(applications.EAI);
+        break;
+      case applications.BACKEND:
+        await generateJavaInteractiveCli(applications.BACKEND);
+        break;
+      case applications.FRONTEND:
+        await generateFrontendInteractiveCli();
+        break;
+    }
+    if (result != EXIT) {
+      await projectConfiguration();
+    } else {
+      await generateDefaultCliForDocsAndStack();
+    }
+  });
+}
+
+/**
+ * interactive CLI for the inputs, needed to generate the backend or eai
+ * @param application - java application that should be generated
+ */
+async function generateJavaInteractiveCli(application: string) {
+  hasJavaApplicationBeenGenerated = true;
+  const groupId = await input({
+    message:
+      "Define value for property groupId (should match expression '^de\\.muenchen\\.[a-z0-9]+(\\.[a-z0-9]+)*$'): ",
+    validate(value) {
+      const pass = value.match(/^de\.muenchen\.[a-z0-9]+(\.[a-z0-9]+)*$/g);
+      return pass ? true : "GroupId name not valid";
+    },
+    required: true,
+  });
+  const artifactId = await input({
+    message: "Define value for property artifactId:",
+    required: true,
```

Comment:
what validations would you add here?

Reply:
Something like `[a-z]+([a-z-])*`. But there must be an official syntax requirement

Reply:
The documentation itself is not really helpful 
`is the name of the jar without version. If you created it, then you can choose whatever name you want with lowercase letters and no strange symbols.`

i would probably just go with `([a-z\-])+` because as example they use `maven, commons-math`

Reply:
then you could theoretically do `test-` but would be ok for me